demo/admin/src/products/ProductsGrid.tsx

```diff
@@ -64,56 +65,74 @@ export function ProductsGrid() {
     const sortModel = dataGridProps.sortModel;
     const client = useApolloClient();
     const { data: categoriesData } = useQuery<GQLProductGridCategoriesQuery, GQLProductGridCategoriesQueryVariables>(productCategoriesQuery);
+    const intl = useIntl();
+
+    const dynamicGridVisibilityProps = useDynamicGridVisibilityModel({
```

Comment:
- I would move this into line 64 and spread directly into dataGridProps
- usePersistentColumnState also returns columnVisibilityModel, which is then overwritten by useDynamicGridVisibilityModel - do you think this is a good idea?
    - possible alternative: usePersistentColumnState could directly support this feature
- have you considered moving the settings into GridColDef?

Reply:
> usePersistentColumnState also returns columnVisibilityModel, which is then overwritten by useDynamicGridVisibilityModel - do you think this is a good idea?
> - possible alternative: usePersistentColumnState could directly support this feature

I'm not too familiar with the use cases for `usePersistentColumnState`. 
If `usePersistentColumnState` is generally always used, it would probably make sense to combine the hooks. 
@johnnyomair what do you think? Should this be part of `usePersistentColumnState`?

> have you considered moving the settings into GridColDef?

Do you mean we could define the behavior in the `columns` variable? Something like this?
Wouldn't we have to wrap `DataGird` with a custom component for that? 🤔 

```ts
const columns: GridColDef[] = [
    {
        field: "overview",
        headerName: "Overview",

        showOnlyIn: "compactView",
        // or
        showOnlyIn: "desktopView",
    }
}
```

Reply:
> I'm not too familiar with the use cases for usePersistentColumnState.

It saves ColumnVisibility, PinnedColumns, ColumnDimensions and ColumnOrder in local storage. I think we can and should use it always - I can not think of a reason why not.

> Something like this?

yes, exactly. 

> Wouldn't we have to wrap DataGird with a custom component for that? 

I should be enough to extend `GridColDef` and add our stuff.

Reply:
If we could add this setting to the `GridColDef` and get those values in `usePersistentColumnState` that would probably be perfect. Is there a way to access the `columns` in the hook though? Would we always have to pass in the `columns`?

Reply:
> Maybe using GridApi

`usePersistentColumnState()` is called outside of the `DataGrid` so we cannot use `useGridApiContext()` for example, if that's what you mean.